cluster-agent/pkg/server/server.go

```diff
@@ -0,0 +1,143 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	v1 "github.com/canonical/microk8s/cluster-agent/pkg/api/v1"
+	v2 "github.com/canonical/microk8s/cluster-agent/pkg/api/v2"
+	"github.com/canonical/microk8s/cluster-agent/pkg/middleware"
+)
+
+const (
+	// ClusterAPIV1 is the version 1 of the MicroK8s cluster API.
+	ClusterAPIV1 = "/cluster/api/v1.0"
+	// ClusterAPIV2 is the version 2 of the MicroK8s cluster API.
+	ClusterAPIV2 = "/cluster/api/v2.0"
+)
+
+// NewServer creates a new *http.ServeMux and registers the MicroK8s cluster agent API endpoints.
+func NewServer(timeout time.Duration, apiv1 *v1.API, apiv2 *v2.API) *http.ServeMux {
```

Comment:
IMO this function seems to be doing to many things.
It could be re-written in a different way so each endpoint's error response logic is  more testable